app/Repository/Group/GroupRepository.php

```diff
@@ -136,19 +137,30 @@ private function generateCacheFileName(int $id): string
     {
         return $this->cacheDirectory . '/group/' . $id . '.json';
     }
-	
-	/**
-	 * Update the groups cache.
-	 *
+
+    /**
+     * Update the groups cache.
      *
-	 * @see https://gruenesandbox.webling.ch/api#header-error-status-codes
-	 */
-	public function updateCache(): void
+     * @see https://gruenesandbox.webling.ch/api#header-error-status-codes
+     * @param int|null $rootId
+     * @throws GroupNotFoundException
+     * @throws WeblingAPIException
+     */
+	public function updateCache(int $rootId = null): void
     {
-		// todo: implement this
-		// note: we have to handle php timeouts
+        if($rootId === null) {
+            $rootId = (int) config('app.cache_root_group_id');
+        }
+
+        $rootGroup = $this->get($rootId);
+        $iterator = GroupIterator::createRecursiveGroupIterator($rootGroup, $this, false);
+
+        /** @noinspection PhpUnusedLocalVariableInspection */
+        foreach ($iterator as $group) {
+		    //reset time limit after each group
+		    set_time_limit(60);
```

Comment:
Meisst greifft dann ein höheres time limit, bei Cyon sind das glaubs 5min, dann wird der apache thread abgeschossen. Sollte dies aber schlussendlich das Problem sein, werde ich mit ihnen für eine Lösung schauen.

Reply:
Ok. Wenn sich am Timeout nichts machen lässt, kann ich auch die Funktion umschreiben, damit man den Cache in mehreren Anläufen aktulalisiert. Aber direkt in einem Rutsch wäre schon eleganter. Braucht es in diesem Fall das set_time_limit in der Schlaufe überhaupt noch?

Reply:
Lass mal so. Wir testen es mal und wenn es dann nicht geht, schauen wir, was die beste Lösung ist...